module/documents/item.mjs

```diff
@@ -1188,28 +1195,33 @@ export default class Item5e extends Item {
         "flags.dnd5e.roll": {type: "attack", itemId: this.id },
         speaker: ChatMessage.getSpeaker({actor: this.actor})
       }
-    };
-
-    // Critical hit thresholds
-    rollConfig.critical = this.getCriticalThreshold();
-
-    // Elven Accuracy
-    if ( flags.elvenAccuracy && CONFIG.DND5E.characterFlags.elvenAccuracy.abilities.includes(this.abilityMod) ) {
-      rollConfig.elvenAccuracy = true;
-    }
-
-    // Apply Halfling Lucky
-    if ( flags.halflingLucky ) rollConfig.halflingLucky = true;
+    }, options);
 
-    // Compose calculated roll options with passed-in roll options
-    rollConfig = foundry.utils.mergeObject(rollConfig, options);
+    /**
+     * A hook event that fires before an attack is rolled for an Item.
+     * @function dnd5e.preRollAttack
+     * @memberof hookEvents
+     * @param {Item5e} item                  Item for which the roll is being performed.
+     * @param {D20RollConfiguration} config  Configuration data for the pending roll.
+     * @returns {boolean}                    Explicitly return false to prevent the roll from being performed.
+     */
+    if ( Hooks.call("dnd5e.preRollAttack", this, rollConfig) === false ) return;
```

Comment:
It would be useful to pass what kind of ammunition is being used in here and potentially allow it to be mutated.